### ISSUE TITLE
pupils of the eyes is wrong in about 20% of results

### ISSUE THREAD

**Issue by suphamster** [sczhou/CodeFormer#213]
![image](https://user-images.githubusercontent.com/17669188/233723534-46db38be-08fe-4100-8292-33256399db1e.png)

**Comment by YilanWang** [comment]
I'm no the author but this is easy to understand, codeformer is a unified model, it doesn't do any facial components strategies. If you want to do some  details you may try:
1. set param w to a smaller number.
2. use some components enhance, like GFPgan.

**Comment by suphamster** [comment]
> I'm no the author but this is easy to understand, codeformer is a unified model, it doesn't do any facial components strategies. If you want to do some details you may try:
> 
> 1. set param w to a smaller number.
> 2. use some components enhance, like GFPgan.

Thanks, but It's actually weak advice, like if you don't like then use other too. Actually I'm tested this case with different w params and even GFPgan. And all my actions get bad results in this case but sometimes it gets me quite good results.

**Comment by YilanWang** [comment]
yeah……It is a recognized challenge in face restore, If it were just testing, I don't think it would be very helpful. It may need to be retrained, for example, the author provided a way to add these components in his training code (in fact, I guess this training code comes from gfpgan...)
